dl/src/main/scala/com/intel/analytics/bigdl/nn/TimeDistributed.scala

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed to Intel Corporation under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * Intel Corporation licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+
+import scala.reflect.ClassTag
+
+/**
+ * This layer is intended to wrap layers that do not support higher Dimensions.
+ * For instance, the Linear layer do not accept 3D input. The TimeDistributed
+ * Layer can wrap the Linear layer, accept 3D input, and feed a sequence of 2D
+ * Tensor to the wrapped Linear layer by reshaping the first two dimensions.
+ *
```

Comment:
I think you should describe why someone may want to use your layer (i.e., applying the contained module to every temporal slice of the input).